### PR TITLE
Fixed a memory leak in Partitioner.

### PR DESCRIPTION
**Pull request by gcatron**
Summary: Fix a case where a Backend can be allocated and never freed. Wrapped the backend in a smart pointer.

Documentation: NA

Test Plan: Ninja check

### CODE REVIEW

File: lib/Partitioner/Partitioner.cpp [hunk 1]
```diff
@@ -1209,7 +1209,7 @@ llvm::Error Partitioner::QuantizationProfilingPartition(
   FunctionToBackendNameMap funcToBackend;
   funcToBackend = backendBasedPartition(F_, backends, cctx);
   module_->eraseFunction(F_);
-  auto backend = createBackend(profilingBackend);
+  std::unique_ptr<Backend> backend(createBackend(profilingBackend));
```
Comment (rdzhabarov) [inline lib/Partitioner/Partitioner.cpp]: did we exercise this pass in tests? (it's very likely to be reported by asan)

Comment (gcatron) [reply lib/Partitioner/Partitioner.cpp]: So I found it as a result of ASAN complaining when I went to land the BackendTest ported to executionEngine2, not sure how it slipped by before.

Comment (beicy) [reply lib/Partitioner/Partitioner.cpp]: For quantization in Partitioner, the tests are added in ./tests/images/run.sh.  It is should be in CI, right?

Comment (nickgg) [reply lib/Partitioner/Partitioner.cpp]: It is run in CI but not in ASAN.

**Pull request closed without merging**